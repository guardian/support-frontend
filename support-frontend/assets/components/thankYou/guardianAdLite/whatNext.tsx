import { palette } from '@guardian/source/foundations';
import OrderedList from 'components/list/orderedList';
import type { ObserverPrint } from 'pages/[countryGroupId]/components/thankYouComponent';
import BulletPointedList from '../utilityComponents/BulletPointedList';

type WhatNextProps = {
	amount: string;
	startDate?: string;
	isSignedIn?: boolean;
	observerPrint?: ObserverPrint;
};

export function WhatNext({
	amount,
	startDate,
	isSignedIn = false,
	observerPrint,
}: WhatNextProps): JSX.Element {
	const whatNextItems = [
		!observerPrint
			? 'You will receive an email confirming the details of your subscription'
			: '',
		!observerPrint && !!startDate
			? `Your payment of Â£${amount}/month will be taken on ${startDate}`
			: '',
		!observerPrint && isSignedIn
			? 'You can now start reading the Guardian website on all your devices without personalised advertising'
			: '',
		observerPrint
			? 'Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.'
			: '',
		observerPrint ? `Your newspaper will be delivered to your door.` : '',
		observerPrint === 'ObserverSubscriptionCard'
			? 'Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.'
			: '',
	];

	const bulletItems: string[] = whatNextItems.filter((item) => item);

	return (
		<>
			{observerPrint ? (
				<OrderedList items={bulletItems} />
			) : (
				<BulletPointedList items={bulletItems} color={palette.neutral[7]} />
			)}
		</>
	);
}
