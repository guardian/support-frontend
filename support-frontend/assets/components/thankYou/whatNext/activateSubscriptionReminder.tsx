import { css } from '@emotion/react';
import { from, space } from '@guardian/source/foundations';

const paragraphCheckInbox = css`
	font-weight: bold;
	margin-top: ${space[5]}px;
	${from.tablet} {
		margin-top: ${space[6]}px;
	}
`;

export function ActivateSubscriptionReminder(): JSX.Element {
	return (
		<>
			<p>
				While your subscription is ready to use on this device, you will need to
				finish setting up your account to read the Guardian website with
				non-personalised advertising on other devices or browsers. You will also
				be able to manage your subscription.
			</p>
			<p css={paragraphCheckInbox}>
				Check your inbox, find the email with the subject line ‘Complete your
				Guardian account’, and follow the link to set your password.
			</p>
		</>
	);
}
