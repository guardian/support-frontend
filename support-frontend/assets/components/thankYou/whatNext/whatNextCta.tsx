import { css } from '@emotion/react';
import { palette, space } from '@guardian/source/foundations';
import {
	LinkButton,
	SvgArrowRightStraight,
} from '@guardian/source/react-components';
import { sendTrackingEventsOnClick } from 'helpers/productPrice/subscriptions';
import { helpCentreUrl, manageSubsUrl } from 'helpers/urls/externalLinks';

const btnStyleOverrides = css`
	justify-content: left;
	margin-bottom: ${space[5]}px;
`;

const neutralFontColor = css`
	color: ${palette.neutral[0]};
`;

type AddressCtaProp = {
	copy: string;
	address: string;
	hasArrow?: boolean;
};

export function AddressCta({
	address,
	copy,
	hasArrow = false,
}: AddressCtaProp): JSX.Element {
	return (
		<>
			{hasArrow ? (
				<LinkButton
					cssOverrides={btnStyleOverrides}
					href={address}
					size="default"
					icon={<SvgArrowRightStraight />}
					iconSide="right"
					nudgeIcon
					aria-label={copy}
				>
					{copy}
				</LinkButton>
			) : (
				<LinkButton
					cssOverrides={btnStyleOverrides}
					href={address}
					size="default"
					aria-label={copy}
				>
					{copy}
				</LinkButton>
			)}
		</>
	);
}

export const myAccountCta = (
	<a
		css={neutralFontColor}
		href={manageSubsUrl}
		onClick={sendTrackingEventsOnClick({
			id: 'checkout_my_account',
			product: 'Paper',
			componentType: 'ACQUISITIONS_BUTTON',
		})}
	>
		Manage my account
	</a>
);

export const helpCenterCta = (linkCopy: string) => (
	<a
		css={neutralFontColor}
		href={helpCentreUrl}
		onClick={sendTrackingEventsOnClick({
			id: 'checkout_help_centre',
			product: 'Paper',
			componentType: 'ACQUISITIONS_BUTTON',
		})}
	>
		{linkCopy}
	</a>
);
