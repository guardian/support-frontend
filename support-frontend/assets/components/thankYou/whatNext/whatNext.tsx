import { palette } from '@guardian/source/foundations';
import OrderedList from 'components/list/orderedList';
import type {
	ActiveProductKey,
	ActiveRatePlanKey,
} from 'helpers/productCatalog';
import type { ObserverPrint } from 'pages/paper-subscription-landing/helpers/products';
import {
	isGuardianWeeklyGiftProduct,
	isGuardianWeeklyProduct,
	isPrintProduct,
} from 'pages/supporter-plus-thank-you/components/thankYouHeader/utils/productMatchers';
import BulletPointedList from '../utilityComponents/BulletPointedList';
import { helpCenterCta, myAccountCta } from './whatNextCta';

export function WhatNext({
	productKey,
	ratePlanKey,
	amount,
	observerPrint,
	startDate,
	isSignedIn = false,
}: {
	productKey: ActiveProductKey;
	ratePlanKey: ActiveRatePlanKey;
	amount: string;
	startDate?: string;
	isSignedIn?: boolean;
	observerPrint?: ObserverPrint;
}) {
	const isSubscriptionCard = productKey === 'SubscriptionCard';
	const isGuardianAdLite = productKey === 'GuardianAdLite';
	const isGuardianWeekly = isGuardianWeeklyProduct(productKey);
	const isGuardianPrint = isPrintProduct(productKey) && !observerPrint;

	if (isGuardianWeekly) {
		const lookoutForEmailCopy =
			'Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.';
		const manageSubscriptionComponent = (
			<>
				You can manage your subscription by visiting {myAccountCta}. For any
				other queries please visit the {helpCenterCta('Help centre')}.
			</>
		);
		const guardianWeeklyItems = [
			lookoutForEmailCopy,
			'Your magazine will be delivered to your door. Please allow 1 to 7 days after publication date for your magazine to arrive, depending on national post services.',
			manageSubscriptionComponent,
		];
		const guardianWeeklyGiftItems = [
			lookoutForEmailCopy,
			'We’re unable to contact the gift recipient directly - make sure to let them know the gift is on its way',
			<>
				Each copy will be delivered to the gift recipient’s door. Here’s a
				reminder of {helpCenterCta('how home delivery works')}.
			</>,
			manageSubscriptionComponent,
		];

		const isGuardianWeeklyGift = isGuardianWeeklyGiftProduct(
			productKey,
			ratePlanKey,
		);
		return (
			<OrderedList
				items={
					isGuardianWeeklyGift ? guardianWeeklyGiftItems : guardianWeeklyItems
				}
			/>
		);
	}

	if (observerPrint ?? isGuardianPrint) {
		const observerItem =
			'The Observer team will be in touch shortly via email to welcome you. Check your spam folder or add the observer.co.uk domain to your preferred sender list.';

		const subscriptionCardItems = [
			'You should receive an email confirming the details of your subscription, and another email shortly afterwards that contains details of how you can pick up your newspapers from tomorrow.',
			'You will receive your Subscription Card in your subscriber pack in the post, along with your home delivery letter.',
			'Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.',
		];

		const paperItems = [
			'Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.',
			'Your newspaper will be delivered to your door.',
		];

		const baseItems = isSubscriptionCard ? subscriptionCardItems : paperItems;
		const allItems = [...baseItems, observerPrint && observerItem];

		return <OrderedList items={allItems.filter(Boolean)} />;
	}

	if (isGuardianAdLite) {
		const guardianAdLiteItems = [
			'You will receive an email confirming the details of your subscription',
			startDate
				? `Your payment of £${amount}/month will be taken on ${startDate}`
				: '',
			isSignedIn
				? 'You can now start reading the Guardian website on all your devices without personalised advertising'
				: '',
		];

		return (
			<BulletPointedList
				items={guardianAdLiteItems.filter(Boolean)}
				color={palette.neutral[7]}
			/>
		);
	}

	return null;
}
