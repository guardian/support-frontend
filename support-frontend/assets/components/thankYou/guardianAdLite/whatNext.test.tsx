import { render } from '@testing-library/react';
import { WhatNext } from './whatNext';

describe('Summary Ts&Cs Snapshot comparison', () => {
	const signIn: Array<boolean | undefined> = [true, false];
	it.each(signIn)(`summaryTs&Cs render product %s correctly`, (signIn) => {
		const { container } = render(
			<WhatNext
				amount={'12'}
				startDate={'Friday, March 28, 2025'}
				isSignedIn={signIn}
			/>,
		);
		expect(container).toMatchSnapshot();
	});
});
