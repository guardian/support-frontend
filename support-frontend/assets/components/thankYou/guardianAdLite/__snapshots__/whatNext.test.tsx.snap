// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Ts&Cs Snapshot comparison whatNext renders correctly (signedIn=false, ObserverSubscriptionCard list) 1`] = `
<div>
  <ol>
    <li
      class="css-k6etrx"
    >
      Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.
    </li>
    <li
      class="css-k6etrx"
    >
      Your newspaper will be delivered to your door.
    </li>
    <li
      class="css-k6etrx"
    >
      Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.
    </li>
    <li
      class="css-k6etrx"
    >
      The Observer team will be in touch shortly via email to welcome you. Check your spam folder or add the observer.co.uk domain to your preferred sender list.
    </li>
  </ol>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison whatNext renders correctly (signedIn=true, ObserverPaper list) 1`] = `
<div>
  <ol>
    <li
      class="css-k6etrx"
    >
      Look out for an email from us confirming your subscription. It has everything you need to know about how to manage it in the future.
    </li>
    <li
      class="css-k6etrx"
    >
      Your newspaper will be delivered to your door.
    </li>
    <li
      class="css-k6etrx"
    >
      The Observer team will be in touch shortly via email to welcome you. Check your spam folder or add the observer.co.uk domain to your preferred sender list.
    </li>
  </ol>
</div>
`;
